No longer store auth guard.

garret-gunter opened the pull request:
Fix for #35

garret-gunter committed:
No longer store auth guard.

patrickbrouwers commented on src/Blameable/BlameableExtension.php:
```diff
@@ -6,23 +6,17 @@
+     *
      */
-    public function __construct(Guard $auth)
+    public function __construct()
```
Please remove the unused constructor.

patrickbrouwers commented on src/Loggable/LoggableExtension.php:
```diff
@@ -6,23 +6,17 @@
+     *
      */
-    public function __construct(Guard $auth)
+    public function __construct()
```
Please remove unused constructor

patrickbrouwers commented on src/ResolveUserDecorator.php:
```diff
@@ -67,8 +59,8 @@ public function setAnnotationReader($reader)
     {
-        if ($this->auth->check()) {
-            call_user_func([$this->wrapped, $this->userSetterMethod], $this->auth->user());
+        if (app('auth')->guard()->check()) {
```
Can you add a getter for `getAuth()` (`return app('auth')`) please.

garret-gunter committed:
Remove unused constructor.

garret-gunter committed:
Remove unused constructor.

garret-gunter committed:
Add getAuth() method.

Get auth manager with getter rather than directly.

garret-gunter committed:
Fix phpdoc formatting.

garret-gunter committed:
Minor formatting fix.

garret-gunter commented:
Applied the changes.

garret-gunter commented:
Do you see anything else that should be changed?

garret-gunter commented:
Is there anything I can do to help this get pulled in?

patrickbrouwers commented on src/ResolveUserDecorator.php:
```diff
@@ -67,8 +59,8 @@ public function setAnnotationReader($reader)
-        if ($this->auth->check()) {
-            call_user_func([$this->wrapped, $this->userSetterMethod], $this->auth->user());
+        if ($this->getAuth()->guard()->check()) {
+            call_user_func([$this->wrapped, $this->userSetterMethod], $this->getAuth());
```
This changes behaviour no? We used to pass the logged in user here. Now it's passing the auth manager.

garret-gunter replied:
Yes, it does.  That is a typo.  Good call.

patrickbrouwers commented on src/ResolveUserDecorator.php:
```diff
@@ -96,4 +88,13 @@ public function getEventSubscriberClass()
+    *
+    * @return \Illuminate\Auth\AuthManager
+    */
+    protected function getAuth(){
```
Can you fix cs here. Brace on new line (see other methods)

patrickbrouwers commented on src/ResolveUserDecorator.php:
```diff
@@ -96,4 +88,13 @@ public function getEventSubscriberClass()
+    /**
+    * Get current Auth manager.
+    *
+    * @return \Illuminate\Auth\AuthManager
```
`Illuminate\Contracts\Auth\Factory`

garret-gunter committed:
Pass the current user to function

Implement current master branch's behavior.

garret-gunter committed:
Fix code format and phpdoc return type

Use the contract that 'auth' is bound to instead of the default resolved manager.

patrickbrouwers commented on src/ResolveUserDecorator.php:
```diff
@@ -96,4 +88,14 @@ public function getEventSubscriberClass()
+    */
+    protected function getAuth()
+    {
+    	return app('auth');
```
Perhaps better to return the guard here straight away.
`->user()` doesn't exist on `\Illuminate\Contracts\Auth\Factory`. (Probably through a magic call, but rather have it explicit)

garret-gunter replied:
Fixed.  Yes, the default auth manager passes __call() straight to the guard.  I agree that explicit is better.

garret-gunter committed:
Directly return guard instead of auth

patrickbrouwers commented:
Thanks!
